src/main/java/com/linecorp/armeria/client/circuitbreaker/EventCounter.java

```diff
@@ -25,12 +27,18 @@
 
     /**
      * Counts success event
+     *
+     * @return An {@link Optional} containing the current {@link EventCount} if it has been updated, or else
+     * an empty {@link Optional}.
      */
-    void onSuccess();
+    Optional<EventCount> onSuccess();
```

Comment:
I wonder if using an `Optional` here and in `onFailure()` will increase the memory footprint, adding pressure to garbage collector. Thoughts?